src/protocols/radius/encode.c

```diff
@@ -303,14 +319,16 @@ static ssize_t encode_tlv_hdr_internal(uint8_t *out, size_t outlen,
 		vp = fr_cursor_current(cursor);
 	}
 
-	return p - out;
+	return init_remaining - fr_dbuff_remaining(dbuff);
 }
 
-static ssize_t encode_tlv_hdr(uint8_t *out, size_t outlen,
+static ssize_t encode_tlv_hdr(fr_dbuff_t *dbuff,
 			      fr_da_stack_t *da_stack, unsigned int depth,
 			      fr_cursor_t *cursor, void *encoder_ctx)
 {
-	ssize_t			slen;
+	ssize_t	slen;
+	uint8_t	*hdr_field = dbuff->p;
```

Comment:
this is a nitpick: we typically would use `attr` here for radius.  We don't generally use `field` suffixes, as they're pretty much redundant.